NOnion/TorCircuit.fs

```diff
@@ -0,0 +1,227 @@
+﻿namespace NOnion
+
+open System.Security.Cryptography
+open System.Threading
+open System.Threading.Tasks
+
+open NOnion.Cells
+open NOnion.Crypto
+open NOnion.Crypto.Kdf
+open NOnion.Utility
+
+type TorCircuit (guard: TorGuard) =
+    let mutable circuitState: CircuitState = CircuitState.Initialized
+    let controlLock: SemaphoreSlim = new SemaphoreSlim (1)
```

Comment:
seems you always use `1` when creating this instance, maybe it's better to create our own wrapper that does this always, and this way you can make the functions `RunSyncWithSemaphore` and `RunSyncWithSemaphore` non-static.

Reply:
^